packages/mui-base/src/useList/listReducer.test.ts

```diff
@@ -0,0 +1,1171 @@
+      {
+        description: 'happy path',
+        key: 'ArrowUp',
+        initialHighlightedItem: '2',
+        disabledItemFocusable: false,
+        disabledItems: [],
+        disableListWrap: false,
+        expected: '1',
+      },
+      {
+        description: 'skips the disabled item',
+        key: 'ArrowUp',
+        initialHighlightedItem: null,
+        disabledItemFocusable: false,
+        disabledItems: ['5'],
+        disableListWrap: false,
+        expected: '4',
+      },
+      {
+        description: 'skips multiple disabled items',
+        key: 'ArrowUp',
+        initialHighlightedItem: '1',
+        disabledItemFocusable: false,
+        disabledItems: ['4', '5'],
+        disableListWrap: false,
+        expected: '3',
+      },
+      {
+        description: 'skips the disabled items and wraps around',
+        key: 'ArrowUp',
+        initialHighlightedItem: '2',
+        disabledItemFocusable: false,
+        disabledItems: ['1', '4', '5'],
+        disableListWrap: false,
+        expected: '3',
+      },
+      {
+        description: 'focuses the disabled item',
+        key: 'ArrowUp',
+        initialHighlightedItem: null,
+        disabledItemFocusable: true,
+        disabledItems: ['5'],
+        disableListWrap: false,
+        expected: '5',
+      },
+      {
+        description: 'does not wrap around',
+        key: 'ArrowUp',
+        initialHighlightedItem: '1',
+        disabledItemFocusable: false,
+        disabledItems: [],
+        disableListWrap: true,
+        expected: '1',
+      },
+      {
+        description: 'remains on the same item when all the previous are disabled',
+        key: 'ArrowUp',
+        initialHighlightedItem: '3',
+        disabledItemFocusable: false,
+        disabledItems: ['1', '2'],
+        disableListWrap: true,
+        expected: '3',
+      },
+      {
+        description: 'all disabled',
+        key: 'ArrowUp',
+        initialHighlightedItem: null,
+        disabledItemFocusable: false,
+        disabledItems: ['1', '2', '3', '4', '5'],
+        disableListWrap: false,
+        expected: null,
+      },
+      {
+        description: 'all disabled but focusable',
+        key: 'ArrowUp',
+        initialHighlightedItem: null,
+        disabledItemFocusable: true,
+        disabledItems: ['1', '2', '3', '4', '5'],
+        disableListWrap: false,
+        expected: '5',
+      },
+      {
+        description: 'happy path',
+        key: 'PageDown',
+        initialHighlightedItem: null,
+        disabledItemFocusable: false,
+        disabledItems: [],
+        disableListWrap: false,
+        expected: '3',
+      },
+      {
+        description: 'skips the disabled item',
+        key: 'PageDown',
+        initialHighlightedItem: null,
+        disabledItemFocusable: false,
+        disabledItems: ['3'],
+        disableListWrap: false,
+        expected: '4',
+      },
+      {
+        description: 'does not wrap around, no matter the setting',
```

Comment:
I couldn't find any specs or examples regarding wrapping when using <kbd>Page Up</kbd> and <kbd>Page Down</kbd>. I found that VSCode suggestions popup wraps highlight when using arrow keys but not when using page keys and it feels sensible to me.